Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -0,0 +1,89 @@
+using Gordon360.Authorization;
+using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Graph;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    [AllowAnonymous]
+    public class TeamsController : GordonControllerBase
+    {
+        private readonly ITeamService _teamService;
+
+        public TeamsController(ITeamService teamService)
+        {
+            _teamService = teamService;
+        }
+
+        ///<summary>
+        ///Get a Team object by ID number
+        ///</summary>
+        /// <param name="teamID"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{teamID}")]
+        public ActionResult<TeamViewModel> GetTeamByID(int teamID)
+        {
+            var team = _teamService.GetTeamByID(teamID);
+
+            if (team == null)
+            {
+                return NotFound();
+            }
+            return Ok(team);
+        }
+
+        /// <summary>
+        /// Create a new team with the requesting user set to team captain
+        /// </summary>
+        /// <param name="newTeam"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("")]
+        public async Task<ActionResult> CreateTeam(TeamUploadViewModel newTeam)
+        {
+            var username = AuthUtils.GetUsername(User);
+            var teamID = await _teamService.PostTeamAsync(newTeam, username);
+
+            return Ok(teamID);
+        }
```

Comment:
Please have your post routes return the resource that was created, rather than just an ID.

That is best practice for REST APIs and perhaps even a standard (@EjPlatzer would know :) ). Any resource-modifying request should return the entire resource object with the modified/created data. That includes: POST, PUT, PATCH, DELETE (delete returns the deleted object).

Reply:
@bennettforkner Thank you for acknowledging me the standard! It is really helpful.

Reply:
@Kevin-Chu58, I think we should get your Routes in first and I can handle some of the best practice refactoring down the line as a lot of my own routes have this exact issue.